src/lib.rs

```diff
@@ -74,7 +74,7 @@ pub trait Group:
     /// this group.
     ///
     /// This function is non-deterministic, and samples from the user-provided RNG.
-    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self;
+    fn random(rng: impl RngCore) -> Self;
```

Comment:
Will this new implementation support passing in `<R: RngCore + !Sized> &mut R`?

Reply:
Yes https://rust-random.github.io/rand/rand/trait.RngCore.html#foreign-impls
```
impl<'a, R> RngCore for &'a mut R where R: RngCore + ?Sized { .. }
```